### PR TITLE
Clesaec/tcomp 1898 record builder

### PR DESCRIPTION
**Pull request by clesaec**
https://jira.talendforge.org/browse/TCOMP-1898 : ajout sur builder pour dataprep (et processors)
### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant

### CODE REVIEW

**Review by undx — APPROVED** [review]
LGTM
(a question)

File: component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/RecordImpl.java [hunk 1]
```diff
@@ -149,6 +158,64 @@ public Builder with(final Entry entry, final Object value) {
...
+                            .format("Entry '%s' of type %s is not compatible with value of type '%s'", schemaEntry.getName(),
+                                    schemaEntry.getType(), value.getClass().getName()));
+                }
+                Optional<Entry> entry = this.entries
+                        .stream()
+                        .filter((Entry e) -> Objects.equals(e.getName(), name))
+                        .findFirst();
+                if (entry.isPresent()) {
+                    this.entries.remove(entry.get());
+                    this.entries.add(schemaEntry);
```
[the remark this answers is not in the document]
Comment (clesaec) [reply component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/RecordImpl.java:205]: Yes, modify order; is order matter here ?

[the remark this answers is not in the document]
Comment (undx) [reply component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/RecordImpl.java:205]: I think that order may matter... here we do not update but remove and add.

[the remark this answers is not in the document]
Comment (clesaec) [reply component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/record/RecordImpl.java]: done

**Review by undx — APPROVED** [review]
LGTM

**Pull request merged**